crates/circuit/src/packed_instruction.rs

```diff
@@ -628,3 +729,23 @@ impl PackedInstruction {
         }
     }
 }
+
+// When cloning a `PackedInstruction` we will need to delete the old cached
+// `Gate` to avoid having duplicated references to the same python object.
```

Comment:
It might be something other than a `Gate` as well, so I think we should use the term "operation" here instead. And, perhaps it'd be better to say that we "explicitly implement `Clone` to avoid copying the source operation's cached Python op", since the term "delete" may not be specific enough for an unfamiliar reader.